### PR TITLE
fix: imageUrlPath edge-slash normalization (#192)

### LINKED ISSUE #192: fix: imageUrlPath edge-slash normalization (parallel to postsUrlPath)
## Bug

`ContentProcessor.imageUrlPath` (`src/content-processor.ts:119-121`) uses the pre-1.5.0 regex pattern:

\`\`\`ts
return \`/\${this.settings.imageDir.replace(/^static\/?/, "")}\`;
\`\`\`

This has the same class of bugs that `postsUrlPath` had before 1.5.0:

1. **No leading/trailing slash normalization.** `imageDir = "/static/images"` becomes `//images` instead of `/images`. `imageDir = "static/images/"` becomes `/images/` instead of `/images`.
2. **Prefix bleed on `static`-prefixed directories.** `imageDir = "static-assets"` becomes `/-assets` (the regex strips the literal `static` and an optional `/`).
3. **Bare directories yield broken URLs.** `imageDir = "images"` (no `static/` prefix) becomes `/images` which is correct only by accident — but `imageDir = ""` becomes `/` which then silently concatenates into `//img.png` downstream.

## Reference

`postsUrlPath` was hardened in 1.5.0 Task 5 to normalize edge slashes and use a boundary regex that preserves non-`content/` directories:

\`\`\`ts
const dir = this.settings.contentDir
  .replace(/^\/+|\/+$/g, "")
  .replace(/^content(?:\/|$)/, "");
return dir ? \`/\${dir}/\` : "/";
\`\`\`

## Fix

Apply the same pattern to `imageUrlPath`:

\`\`\`ts
private imageUrlPath(): string {
  const dir = this.settings.imageDir
    .replace(/^\/+|\/+$/g, "")
    .replace(/^static(?:\/|$)/, "");
  return dir ? \`/\${dir}/\` : "/";
}
\`\`\`

Note: this changes the trailing behavior — current code returns `/images` (no trailing slash) and image URLs are concatenated as `\${urlPath}/\${filename}`. After this change it returns `/images/` and concatenation shape must change to `\${urlPath}\${filename}`. Update `convertImageReferences` (`src/content-processor.ts:273-285`) accordingly.

## Tests

Parallel to `postsUrlPath` tests. Cases:
- Default `static/images` → `/images/`
- Bare `images` → `/images/`
- `"static"` (bare) → `/`
- `"static-assets"` → `/static-assets/` (boundary preserved)
- `"/static/images/"` → `/images/`
- `"staticfiles/img"` → `/staticfiles/img/` (no bleed)

## History

Flagged during 1.5.0 wrap-up as a parallel fix but deferred (see `TODO.md` follow-ups). Surfaces as broken image URLs in published posts for any user with a non-default `imageDir`.

### PR DESCRIPTION
**Pull request by philoserf**
## Summary

- Apply the `postsUrlPath` boundary-regex pattern to `ContentProcessor.imageUrlPath`: normalize edge slashes, strip `static` only at a word boundary, always return a trailing-slash-terminated path.
- Migrate the trailing slash from the concat site in `convertImageReferences` (`${urlPath}/${sanitizedName}` → `${urlPath}${sanitizedName}`) — net output unchanged for common cases.
- Five new tests covering previously-broken configs: bare `static`, `static-assets`, `staticfiles/img`, `/static/images/`, empty `imageDir`. Default and `assets/img` cases remain covered by pre-existing tests (content-processor.test.ts:403-419).

## Why it matters

Users with `imageDir` values like `static-assets` or `staticfiles/img` saw broken image URLs (`/-assets/hero.png`, `/files/img/hero.png`) because the old `^static\/?` regex stripped the literal `static` without a word boundary. Edge slashes (`/static/images/`) produced double-slash URLs. Flagged during 1.5.0 wrap-up as a parallel fix to what shipped for `postsUrlPath`, deferred, now done.

## Test plan

- [x] `bun run check` clean (typecheck + biome)
- [x] `bun test` — 234 pass (229 → 234; added 5 edge-case tests)
- [x] `bun run build.ts` clean (125.3 KB)
- [x] All 18 existing image URL assertions continue to pass unchanged — verified by algebra of the new regex against each input.

Closes #192

### CODE REVIEW

File: src/content-processor.test.ts [hunk 1]
```diff
@@ -418,6 +418,51 @@ describe("Image reference conversion", () => {
     expect(result.content).toContain("![hero.png](/assets/img/hero.png)");
   });
 
+  test("imageUrlPath: bare 'static' collapses to root", () => {
+    const cp2 = makeProcessor({ imageDir: "static" });
+    const result = cp2.process(
+      wrap("title: Test\nstatus: publish", "![[hero.png]]"),
+      "test.md",
```
[the remark this answers is not in the document]
Comment (philoserf) [reply src/content-processor.test.ts:425]: Good catch — miscounted in the PR description. The diff adds 5 new cases (229 → 234), not 6. The default `static/images` case is already covered by `content-processor.test.ts:373` and the no-static-prefix case by line 418, so I deliberately didn't duplicate them. Updated the PR description to say 5 new tests.
